### PR TITLE
Fix the bug where ListSnapshot failing to query deleted fs

### LINKED ISSUE #201: EFS SnapshotsList returns error if originating filesystem is deleted
The problem happens in case:
- A snapshot is taken from filesystem.
- Filesystem is deleted
- SnapshotsList returns an error saying that the filesystem cannot be found.

### PR DESCRIPTION
**Pull request by hakanmemisoglu**
Fixes the problem where EFS ListSnapshots fails to query already deleted filesystem.

* Do not fail when VolumeGet operation in ListSnapshots returns an error.
* Do not fail when volume argument in snapshotFromRecoveryPointByVault is nil.

Fixes #201

### CODE REVIEW

File: pkg/blockstorage/awsefs/awsefs.go [hunk 1]
```diff
@@ -453,10 +453,9 @@ func (e *efs) snapshotsFromRecoveryPoints(ctx context.Context, rps []*backup.Rec
 		if err != nil {
 			return nil, errors.Wrap(err, "Failed to get volume ID from recovery point ARN")
 		}
-		vol, err := e.VolumeGet(ctx, volID, "")
-		if err != nil {
-			return nil, errors.Wrap(err, "Failed to get EFS volume")
-		}
+		// VolumeGet might return error since originating filesystem might have
+		// been deleted.
+		vol, _ := e.VolumeGet(ctx, volID, "")
```
Comment (tdmanv) [inline pkg/blockstorage/awsefs/awsefs.go]: is it possible to check the specific NotFound error?

File: pkg/blockstorage/awsefs/conversion.go [hunk 2]
```diff
@@ -90,8 +94,9 @@ func snapshotFromRecoveryPointByVault(rp *backup.RecoveryPointByBackupVault, vol
 	if rp.RecoveryPointArn == nil {
 		return nil, errors.New("Recovery point has no RecoveryPointArn")
 	}
-	if volume == nil {
-		return nil, errors.New("Nil volume as argument")
+	encrypted := false
+	if volume != nil {
+		encrypted = volume.Encrypted
 	}
```
Comment (pavannd1) [inline pkg/blockstorage/awsefs/conversion.go]: What happens when the volume is `nil` since we are removing this check?

Comment (hakanmemisoglu) [reply pkg/blockstorage/awsefs/conversion.go]: `blockstorage.Snasphot`'s `Volume` field will be `nil`, `Encrypted` field will be `false`. 
Do you see any panic situation?

Comment (pavannd1) [reply pkg/blockstorage/awsefs/conversion.go]: I don't see any usage of `blockstorage.Snapshot.Volume` for EFS but want to make sure it's okay to set it to `nil`

**Pull request merged**